File: 100-Days-Series/Day 16/DecimalToBinary.c
#include<stdio.h>
// Function for Decimal to Binary conversion
long getDecimalToBinary(int decimal){
    long long binary = 0;
    int remainder,place=1;
    while(decimal!=0){
        remainder=decimal%2; // Get the last binary digit
        binary+=remainder*place; // Add it to the binary result
        decimal/=2;     // Divide the number by 2
        place*=10;      // Move to the next place value
    }
    return binary;
}
// Let's Start here
int main(){
    int decimal;
    printf("\nEnter Decimal number : ");
    scanf("%d",&decimal);
    long long binary=getDecimalToBinary(decimal);
    printf("Binary Equivalent : %lld",binary);
    printf("\n%d",1%2);
    return 0;
}// Note:   1 % 2 = 1;

File: 100-Days-Series/Day 16/DecimalToBinary.java
public class DecimalToBinary {
    // Function for Decimal to Binary conversion
    public static long getDecimalToBinary(int decimal){
        int place=1;
        long binary=0;
        while(decimal!=0){
            int remainder=decimal%2;  //Get the last binary digit          
            binary+=remainder*place;  //  Add it to the binary result
            decimal/=2;         // Divide the number by 2
            place*=10;      //Move to the next place value
        }
        return binary;              
    }
    // Main Method
    public static void main(String [] args){
        java.util.Scanner s1=new java.util.Scanner(System.in);
        System.out.print("Enter Decimal Number : ");
        int decimal=s1.nextInt();
        long binary=getDecimalToBinary(decimal);
        System.out.println("Binary Equivalent is : "+binary);
    }
}
